src/consensus/consensus.h

```diff
@@ -15,7 +15,7 @@ static const unsigned int BIP102_FORK_BUFFER = (144 * 90);
 /** The maximum allowed size for a serialized block, in bytes (only for buffer size limits) */
 static const unsigned int MAX_BLOCK_SERIALIZED_SIZE = (8 * 1000 * 1000);
 /** The maximum allowed weight for a block, see BIP 141 (network rule) */
-static const unsigned int MAX_BLOCK_WEIGHT = 4000000;
+static const unsigned int MAX_BLOCK_WEIGHT = (8 * 1000 * 1000);
```

Comment:
@jgarzik 
I've thought about it some more. While the largest SegWit block I've seen in testing was only ~3.7MB, it may be possible to create a complex enough series of transactions to create a SegWit block larger than 4MB. If that were to happen with the higher 8M limit, the result would be considered an invalid block by all other SegWit nodes that are not SegWit2x. For that reason, I'd personally ere on the side of caution and keep the limit at 4M with the softfork, only increasing to 8M with the hardfork.